src/web_interface/components/analysis_routes.py

```diff
@@ -69,22 +66,44 @@ def _show_analysis_results(self, uid, selected_analysis=None, root_uid=None):
-        view = self._get_analysis_view(selected_analysis) if selected_analysis else get_template_as_string('show_analysis.html')
         with ConnectTo(InterComFrontEndBinding, self._config) as sc:
             analysis_plugins = sc.get_available_analysis_plugins()
-        return render_template_string(view,
-                                      uid=uid,
-                                      firmware=file_obj,
-                                      selected_analysis=selected_analysis,
-                                      all_analyzed_flag=included_fo_analysis_complete,
-                                      summary_of_included_files=summary_of_included_files,
-                                      root_uid=root_uid,
-                                      firmware_including_this_fo=self._get_firmware_ids_including_this_file(file_obj),
-                                      analysis_plugin_dict=analysis_plugins,
-                                      other_versions=other_versions,
-                                      uids_for_comparison=get_comparison_uid_list_from_session(),
-                                      user_has_admin_clearance=user_has_privilege(current_user, privilege='delete'),
-                                      known_comparisons=known_comparisons)
+        return render_template_string(
+            self._get_analysis_view(selected_analysis) if selected_analysis else get_template_as_string('show_analysis.html'),
+            uid=uid,
+            firmware=file_obj,
+            selected_analysis=selected_analysis,
+            all_analyzed_flag=included_fo_analysis_complete,
+            summary_of_included_files=summary_of_included_files,
+            root_uid=none_to_none(root_uid),
+            firmware_including_this_fo=self._get_firmware_ids_including_this_file(file_obj),
+            analysis_plugin_dict=analysis_plugins,
+            other_versions=other_versions,
+            uids_for_comparison=get_comparison_uid_list_from_session(),
+            user_has_admin_clearance=user_has_privilege(current_user, privilege='delete'),
+            known_comparisons=known_comparisons,
+            available_plugins=self._get_used_and_unused_plugins(
+                file_obj.processed_analysis,
+                [x for x in analysis_plugins.keys() if x != 'unpacker']
+            )
+        )
+
+    def _start_single_file_analysis(self, uid):
+        if user_has_privilege(current_user, privilege='submit_analysis'):
+            with ConnectTo(FrontEndDbInterface, self._config) as database:
+                file_object = database.get_object(uid)
+            file_object.scheduled_analysis = request.form.getlist('analysis_systems')
+            with ConnectTo(InterComFrontEndBinding, self._config) as intercom:
+                intercom.add_single_file_task(file_object)
+        else:
+            flash('You have insufficient rights to add additional analysis')
```

Comment:
`... to add additional analysis`
typo? an additional analysis/additional analyses?